script.js

```diff
@@ -1,28 +1,86 @@
-var values  = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A"];
+var values  = [2, 3, 4, 5, 6, 7, 8, 9, 10, "Jack", "Queen", "King", "Ace"];
 var suits   = ["Clubs", "Diamonds", "Hearts", "Spades"];
 var game = {
   deck: [],
   players: [],
   hands: [],
   buildDeck: function(){
-
+      for (var myRank = 0; myRank < values.length; myRank++) {
+        for (var mySuit = 0; mySuit < suits.length; mySuit++) {
+          this.deck.push({ "rank" : values[myRank], "suit" : suits[mySuit]});
+        }
+      }
   },
   shuffleDeck: function(){
+    var currentIndex = this.deck.length, temporaryValue, randomIndex;
+
+    // While there remain elements to shuffle...
+    while (0 !== currentIndex) {
 
+      // Pick a remaining element...
+      randomIndex = Math.floor(Math.random() * currentIndex);
+      currentIndex -= 1;
+
+      // And swap it with the current element.
+      temporaryValue = this.deck[currentIndex];
+      this.deck[currentIndex] = this.deck[randomIndex];
+      this.deck[randomIndex] = temporaryValue;
+    }
   },
   getMorePlayers: function(){
-
+    var message = "Enter the name of a new player.";
+    var defaultText = "Name";
+    var name = "";
+    while (this.players.length < 2) {
+      name = prompt(message,defaultText);
+      this.players.push(name);
+    }
+    message += "\nLeave blank to stop adding players."
+    while (name = prompt(message,"")) {
```

Comment:
I appreciate that it's "bad form" and all, but I wanted to see if it worked in JS. And it's **_cool_**. (Yeah, I know, it's cool until someone has to maintain it.)